volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -258,6 +258,18 @@ def get_threads(self) -> Iterable[interfaces.objects.ObjectInterface]:
         ):
             yield task
 
+    def get_pid(self) -> int:
+        """Returns the pid of this process"""
+        return self.tgid
+
+    def get_parent_pid(self) -> int:
+        """Returns the pid of parent of this process"""
+        return self.parent.get_pid() if self.parent else 0
```

Comment:
This is something I have in my TO DO list. We should use `task_struct.real_parent` instead for this. For instance, see the [getppid() ](https://elixir.bootlin.com/linux/v6.10.9/source/kernel/sys.c#L989) syscall. A task's `parent` and `real_parent` can differ, for instance when one task is ptracing another. In this case, the `parent` refers to the tracer, while the `real_parent` remains the original (real) parent task. I know we use `task.parent` for this purpose everywhere in the framework, but that needs to be changed. If you are going to replace all instances of `task.parent.tgid` (or `task.parent.pid` pending discussion https://github.com/volatilityfoundation/volatility3/issues/981) with `get_parent_pid()`, I think this is a good opportunity to also fix this issue.

Reply:
Nice -  I did not know that - I've done this now.